File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/guardosphere/GuardianDetailWidget.java
package de.limbusdev.guardianmonsters.fwmengine.guardosphere;

import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.scenes.scene2d.Group;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
import com.badlogic.gdx.utils.Align;

import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian;
import de.limbusdev.guardianmonsters.services.Services;

/**
 * GuardianDetailWidget
 *
 * @author Georg Eckert 2017
 */

public class GuardianDetailWidget extends Group {

    private static final int WIDTH = 152;
    private static final int HEIGHT= 180;

    private Image monsterSprite;
    private Label name;
    private Label level;

    public GuardianDetailWidget(Skin skin) {
        setSize(WIDTH,HEIGHT);
        Image background = new Image(skin.getDrawable("guardosphere-frame"));
        background.setSize(WIDTH,HEIGHT);
        background.setPosition(0,0, Align.bottomLeft);
        addActor(background);

        monsterSprite = new Image(skin.getDrawable("transparent"));
        monsterSprite.setSize(128,128);
        monsterSprite.setPosition(12,HEIGHT-8,Align.topLeft);
        addActor(monsterSprite);

        name = new Label("Name", skin, "white");
        name.setSize(92,20);
        name.setPosition(12,20,Align.bottomLeft);
        addActor(name);

        level = new Label("Lvl 0", skin, "white");
        level.setSize(32,20);
        level.setPosition(12+96,20,Align.bottomLeft);
        addActor(level);
    }

    public void showDetails(AGuardian guardian) {
        TextureAtlas.AtlasRegion region = Services.getMedia().getMonsterSprite(guardian.getSpeciesDescription().getID());
        monsterSprite.setDrawable(new TextureRegionDrawable(region));
        name.setText(guardian.getNickname());
        level.setText("Lvl " + guardian.getStatistics().getLevel());
    }
}

File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/guardosphere/GuardianDetailWidget.kt
package de.limbusdev.guardianmonsters.fwmengine.guardosphere

import com.badlogic.gdx.scenes.scene2d.Group
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Skin
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable
import com.badlogic.gdx.utils.Align

import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian
import de.limbusdev.guardianmonsters.scene2d.lSetPosition
import de.limbusdev.guardianmonsters.scene2d.lSetSize
import de.limbusdev.guardianmonsters.services.Services
import ktx.actors.plus

/**
 * GuardianDetailWidget
 *
 * @author Georg Eckert 2017
 */
class GuardianDetailWidget(skin: Skin) : Group()
{
    private val monsterSprite: Image
    private val name: Label
    private val level: Label

    init
    {
        setSize(WIDTH, HEIGHT)

        val background = Image(skin.getDrawable("guardosphere-frame"))
        background
                .lSetSize(WIDTH, HEIGHT)
                .lSetPosition(0f, 0f, Align.bottomLeft)

        monsterSprite = Image(skin.getDrawable("transparent"))
        monsterSprite
                .lSetSize(128f, 128f)
                .lSetPosition(12f, HEIGHT - 8f, Align.topLeft)

        name = Label("Name", skin, "white")
        name
                .lSetSize(92f, 20f)
                .lSetPosition(12f, 20f, Align.bottomLeft)

        level = Label("Lvl 0", skin, "white")
        level
                .lSetSize(32f, 20f)
                .lSetPosition(12f + 96f, 20f, Align.bottomLeft)

        this+background
        this+monsterSprite
        this+name
        this+level
    }

    fun showDetails(guardian: AGuardian?)
    {
        if(guardian != null)
        {
            val region = Services.getMedia().getMonsterSprite(
                    guardian.speciesDescription.ID,
                    guardian.abilityGraph.currentForm)
            monsterSprite.drawable = TextureRegionDrawable(region)
            name.setText(Services.getL18N().getGuardianNicknameIfAvailable(guardian))
            level.setText("Lvl $guardian.individualStatistics.level")
        }
    }

    companion object
    {
        private const val TAG = "GuardianDetailWidget"
        private const val WIDTH = 152f
        private const val HEIGHT = 180f
    }
}
